### PR TITLE
Projectile and damage events small tweak

### PR DESCRIPTION
**Pull request by PixelTheKermit**
## About the PR

- Made ProjectileHitEvent relay the damage and shooter
- Made DamageModifyEvent and BeforeDamageChangedEvent relay the origin of the damage

## Why / Balance

I need this downstream, and I thought PRing it upstream would hopefully mitigate any conflicts that might arise in the future.

I would need to know the origin of the damage to validate if the attack was made by an actual entity or not, and if the entity was the player itself.

I would need ProjectileHitEvent to relay the damage so I can modify it with other components.

## Technical details

This hopefully shouldn't break anything as it doesn't actually change all that much. (However if it does, then uh... whoops!)
This could also be helpful to upstream and maybe other downstreams.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

What are changelogs?

### CODE REVIEW

File: Content.Server/Projectiles/ProjectileSystem.cs [hunk 1]
```diff
@@ -43,12 +43,12 @@ private void OnStartCollide(EntityUid uid, ProjectileComponent component, ref St
...
         }
 
-        var ev = new ProjectileHitEvent(target);
+        var ev = new ProjectileHitEvent(component.Damage, target, component.Shooter);
         RaiseLocalEvent(uid, ref ev);
 
         var otherName = ToPrettyString(target);
         var direction = args.OurBody.LinearVelocity.Normalized();
-        var modifiedDamage = _damageableSystem.TryChangeDamage(target, component.Damage, component.IgnoreResistances, origin: component.Shooter);
+        var modifiedDamage = _damageableSystem.TryChangeDamage(target, ev.Damage, component.IgnoreResistances, origin: component.Shooter);
```
Comment (PrPleGoo) [inline Content.Server/Projectiles/ProjectileSystem.cs:51]: Do you expect the damage value to have been changed by the raising of the event?

Comment (PixelTheKermit) [reply Content.Server/Projectiles/ProjectileSystem.cs:51]: huh?

Comment (PixelTheKermit) [reply Content.Server/Projectiles/ProjectileSystem.cs:51]: Oh uh, I think I know what you mean.
I'd like the damage value to be changed by the systems that are subscribed to that event, if that's what you mean.

Comment (PrPleGoo) [reply Content.Server/Projectiles/ProjectileSystem.cs:51]: I'm pretty sure the event raising has no way to guarantee an order of operation. So *0.5 and -2 within two separate subscriptions will have an unpredictable outcome.

Maybe put whatever changes you want to do on the event's fields and guarantee the correct order of operations here?

Comment (PixelTheKermit) [reply Content.Server/Projectiles/ProjectileSystem.cs:51]: Ah I see what you mean.
I'll work on resolving that.

Comment (PixelTheKermit) [reply Content.Server/Projectiles/ProjectileSystem.cs:51]: > I'm pretty sure the event raising has no way to guarantee an order of operation. So *0.5 and -2 within two separate subscriptions will have an unpredictable outcome.

So I looked a bit into it and what I found is that people would use `DamageSpecifier.ApplyModifierSet` anyways, which shouldn't cause too many issues with the order at which stuff is done.

**Pull request merged**